### DISCUSSION TITLE
shm_open for android [14806]

### DISCUSSION THREAD

**Discussion by Evast1995** [eProsima/Fast-DDS#2728] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

。

### Current behavior

Currently, FastDDS uses Boost to implement shared memory, but Android does not support System V IPC
/* SysV semaphores*/
/* SysV shared memory segments */
/* SysV message queues */
/* General IPC definitions */
https://android.googlesource.com/platform/ndk/+/4e159d95ebf23b5f72bb707b0cb1518ef96b3d03/docs/system/libc/SYSV-IPC.TXT

### Steps to reproduce

。

### Fast DDS version/commit

Version: 2.6.0

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Shared Memory Transport (SHM)

**Comment by JLBuenoLopez** [comment]
Hi @Evast1995,

Android is not an [officially supported platform](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md#platform-support) so I am going to move this issue to the proper discussion forum according to the project [CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#unofficial-support).

**Comment by jsan-rt** [comment]
Hi @Evast1995,

You are correct. `shm_open` and `shm_unlink` are not implemented in Android. 

Boost checks for several available capabilities, `_POSIX_SHARED_MEMORY_OBJECTS` being one of them. This is defined in the NDK as `__BIONIC_POSIX_FEATURE_MISSING (-1)`.
What Android does support is `mmap/msync/munmap` (you can see it in the NDK being defined as `#define _POSIX_MAPPED_FILES _POSIX_VERSION /* mmap/msync/munmap. */` ) so Boost will fall back to memory mapped files.
